test/integration_test.py

```diff
@@ -2054,6 +2055,14 @@ def test_response_security_headers(self):
             '/oauth2_redirect': {'Cache-Control': 'no-store'},
             '/health/basic': {'Cache-Control': 'no-store'}
         }
+        # https://www.w3.org/TR/CSP2/#policy-syntax
+        directive_re = re.compile(r'[ \t]*([a-zA-Z0-9-]+)'
+                                  # Space, tab and any visible character
+                                  # (0x21-0xFE) except for comma (0x2C) or
+                                  # semicolon (0x3B).
+                                  r'(?:[ \t]([ \t\x21-\x2B\x2D-\x3A\x3C-\xFE]+))?')
```

Comment:
```suggestion
                                  r'(?:[ \t]([ \t\x21-\x2B\x2D-\x3A\x3C-\xFE]+))?')
```

The grammar says that the value can be empty, but you use `+` and I didn't understand your reasoning given in your most recent comment on this. Please either use `*` or raise in PL.